daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala

```diff
@@ -852,56 +856,63 @@ private[lf] object Speedy {
   private[speedy] sealed trait Kont {
 
     /** Execute the continuation. */
-    def execute(v: SValue, machine: Machine): Unit
+    def execute(v: SValue): Unit
   }
 
   /** Final continuation; machine has computed final value */
-  private[speedy] final case object KFinished extends Kont {
-    def execute(v: SValue, machine: Machine) = {
+  private[speedy] final case class KFinished(machine: Machine) extends Kont {
+    def execute(v: SValue) = {
       if (enableInstrumentation) {
         machine.track.print()
       }
       throw SpeedyHungry(SResultFinalValue(v))
     }
   }
 
-  private[speedy] final case class KOverApp(
-      savedBase: Int,
-      newArgs: Array[SExprAtomic],
-      frame: Frame,
-      actuals: Actuals)
+  private[speedy] final case class KOverApp(machine: Machine, newArgs: Array[SExprAtomic])
```

Comment:
I'm slightly troubled by the idea of storing pointers to mutable objects in long-living objects. It's probably fine in this instance but I'm not sure. @remyhaemmerle-da do you have any thoughts on this?

Reply:
So specifically: I think you referring to references from the continuations to the machine. ?
This doesn't worry me.
I sure hope the machine is longer lived that the continuations that live in it.

Reply:
Are there any more thoughts on this issue?

Reply:
I've decided I'm ok with this but would really like to hear from @remyhaemmerle-da about this too.